kernel/src/log_segment/tests.rs

```diff
@@ -107,6 +118,100 @@ fn build_log_with_paths_and_checkpoint(
     (Box::new(client), log_root)
 }
 
+// Create an in-memory store and return the store and the URL for the store's _delta_log directory.
+fn new_in_memory_store() -> (Arc<InMemory>, Url) {
+    (
+        Arc::new(InMemory::new()),
+        Url::parse("memory:///")
+            .unwrap()
+            .join("_delta_log/")
+            .unwrap(),
+    )
+}
+
+// Writes a record batch obtained from engine data to the in-memory store at a given path.
+fn write_record_batch_to_store(
```

Comment:
I think it's more important to highlight that what we're writing here is parquet, so maybe `write_parquet_to_store`